### DISCUSSION TITLE
Melhor prática para eventos Javascript

### DISCUSSION THREAD

**Discussion by ytalobessa** [frontendbr/forum#181] [Perguntas e Respostas]
Glra, estou estudando Javascript e cheguei na parte dos eventos, me deparei com três formas de Syntax do mesmo evento, nesse caso o duplo clique, qual dessas formas seria a mais adequada? ou melhor pratica?

link do w3scholl: http://www.w3schools.com/jsref/event_ondblclick.asp

**Comment by luisfmsouza** [comment]
Sempre que possível eu uso `object.addEventListener`

**Comment by napalmdev** [comment]
Eu prefiro sempre o addEventListener atacando os Data-Attributes, pois posteriormente vc pode remover os ouvintes caso necessário com removeEventListener.

**Comment by fdaciuk** [comment]
Comece trocando o w3schools pela MDN (ou o https://devdocs.io) =)
Boa prática é sempre utilizar o `addEventListener`, mas é importante conhecer as outras formas de atribuir eventos. Dificilmente você vai fazer isso diretamente no HTML.

Com o `addEventListener` você pode usar o `removeEventListener` posteriormente para remover o evento, e pode atrelar o mesmo evento mais de uma vez.

E com o `object.ondblclick = function () {}`, você atribui o evento somente uma vez ao objeto. A última atribuição sempre substitui as anteriores =)

**Comment by ytalobessa** [comment]
Blz, pessoal, vlw as dicas, já troquei o W3S pela MDN =)

**Comment by fdaciuk** [comment]
Show @ytalobessa! Pode deixar a issue aberta, outras pessoas podem ter a mesma dúvida ;)
